game-core/src/main/java/games/strategy/triplea/ui/DicePanel.java

```diff
@@ -27,59 +26,82 @@
   public DicePanel(final UiContext uiContext, final GameData data) {
     this.uiContext = uiContext;
     this.data = data;
-    setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
+    setLayout(new GridBagLayout());
+    setBorder(BorderFactory.createEmptyBorder(8, 16, 0, 0));
   }
 
   void setDiceRollForBombing(final List<Die> dice, final int cost) {
     removeAll();
+    add(new JLabel("Cost: " + cost, SwingConstants.LEFT));
     add(create(dice));
-    add(Box.createVerticalGlue());
-    add(new JLabel("Cost:" + cost));
-    invalidate();
+    addBottomLabel(new JLabel());
   }
 
   /** Sets the dice roll to display. */
   public void setDiceRoll(final DiceRoll diceRoll) {
-    SwingAction.invokeNowOrLater(
-        () -> {
-          removeAll();
-          for (int i = 1; i <= data.getDiceSides(); i++) {
-            final List<Die> dice = diceRoll.getRolls(i);
-            if (dice.isEmpty()) {
-              continue;
-            }
-            add(new JLabel("Rolled at " + i + ":"));
-            add(create(diceRoll.getRolls(i)));
-          }
-          add(Box.createVerticalGlue());
-          add(new JLabel("Total hits:" + diceRoll.getHits()));
-          validate();
-          invalidate();
-          repaint();
-        });
+    removeAll();
+    add(
+        new JLabel(
+            "<html><b><font style='font-size:120%'>Total hits: <font color='#8B0000'>"
+                + diceRoll.getHits(),
+            SwingConstants.LEFT));
+    add(new JSeparator());
+
+    for (int i = 1; i <= data.getDiceSides(); i++) {
+      final List<Die> dice = diceRoll.getRolls(i);
+      if (dice.isEmpty()) {
+        continue;
+      }
+      add(makeDiceRolledLabel(dice, i));
+      add(create(dice));
+    }
+    addBottomLabel(new JLabel());
+    invalidate();
+    validate();
+    repaint();
+  }
+
+  private JLabel makeDiceRolledLabel(final List<Die> dice, final int value) {
+    int hits = 0;
+    for (final Die die : dice) {
+      if (die.getType() == Die.DieType.HIT) {
+        hits++;
+      }
+    }
```

Comment:
If you want to got for streams
```suggestion
    final long hits = dice.stream().map(Die::getType).filter(Die.DieType.HIT::equals).count();
```

Reply:
Done. Thanks!